packages/typescript/src/transactions/Transaction.ts

```diff
@@ -383,15 +391,129 @@ export class Transaction {
 		return this.object(Inputs.SharedObjectRef(...args));
 	}
 
+	#fork() {
+		const fork = new Transaction();
+		fork.#data = this.#data.shallowClone();
+		fork.#parent = {
+			transaction: this,
+			commandIndex: this.#data.commands.length,
+			inputIndex: this.#data.inputs.length,
+		};
+		fork.#lastPendingTask = this.#lastPendingTask;
+		return fork;
+	}
+
+	#commit(transaction: Transaction) {
+		this.#data = transaction.#data;
+		this.#lastPendingTask = transaction.#lastPendingTask;
+	}
+
+	/** Destructively merges a transaction into the current transaction. The passed transaction will not be usable after merging. */
```

Comment:
> The passed transaction will not be usable after merging

Outside of the scope of this PR, but at some point I really wanted to have a way to "poison" a transaction so that after you passed it to methods like signAndExecute it was unusable, to avoid issues with folks re-using the same TX object and accidentally equivocating objects. Would also be useful for a case like this where after merging we'd be able to mark the passed transaction as unusable.